### ISSUE TITLE
build(dev-container): add gpg signing key support

### ISSUE THREAD

**Issue by petermetz** [hyperledger-cacti/cacti#2209]
### Description
As a contributor or maintainer, I want to have my git signing keys transferred to the dev container automatically if possible so that it's much less hassle to rebuild a dev container from scratch.

https://code.visualstudio.com/docs/devcontainers/containers#_sharing-git-credentials-with-your-container

### Acceptance Criteria
1. A full rebuild or a first time build of the dev container does not need manual steps from the contributor in order to have their git workflow operate as usual (with the assumption that they have their host-OS side of things set up as described by the guide linked above)
2. Git (GitHub) credentials are taken care of
3. Git username + email and signing keys are transferred (this is working as intended as of right now but adding it here in the checklist in case the solution would otherwise break it)
4. Git gpg signing keys transferred as well
5. Big picture is that after starting the dev container, git push/pull/commit commands should work as expected.
6. Solution tested on macOS, Windows (WSL 2) and Ubuntu Linux 20.04 and 22.04

**Comment by solo-daemon** [comment]
@petermetz i looked into this issue a mount of the .gnupg directory looks like an idea but then the developer will have to locate to their .gnupg directory before building the container. But that will not exactly be automating it. What do you think?

**Comment by petermetz** [comment]
> @petermetz i looked into this issue a mount of the .gnupg directory looks like an idea but then the developer will have to locate to their .gnupg directory before building the container. But that will not exactly be automating it. What do you think?

@solo-daemon I agree that's not ideal unfortunately. In the meantime I also have some more data points to share which probably just makes the issue even harder/more complicated because it hints at the problem being heavily host OS specific: It does not work on my NixOS setup, no matter what I try but it does work on my PopOS laptop without having done anything, so I'm a bit stumped about how we should approach this.
